core/vm/instructions.go

```diff
@@ -431,11 +426,11 @@ func touchEachChunksAndChargeGas(offset, size uint64, address []byte, contract *
 		index[31] = subIndex
 
 		var value []byte
-		if contract != nil {
+		if code != nil {
```

Comment:
We should also catch the case in which the code is an empty slice. It's technically covered by the for loop afterwards, still it's better to future-proof it.

```suggestion
		if len(code) > 0 {
```

Reply:
Okay, I added it in this PR:  `if code != nil && len(code) > 0 {`